src/Diagnostics.DataProviders/TokenService/TokenServiceBase.cs

```diff
@@ -99,6 +99,10 @@ public async Task<string> GetAuthorizationTokenAsync()
         {
             if (!tokenAcquiredAtleastOnce)
             {
+                while (acquireTokenTask == null)
+                {
+                    await Task.Delay(1000);
```

Comment:
So we want to wait 1s before acquiring token the first time?

Reply:
Yes, it's much faster than 20 seconds. But obviously there is a better way to implement here by using a proper synchronization object. Let me improve so it makes more sense.

Reply:
Replaced Task.Delay() with ManualResetEvent.WaitOne()